### ISSUE TITLE
package.json contains invalid range and fails to resolve http-parser-js

### ISSUE THREAD

**Issue by mgroenhoff** [faye/websocket-driver-node#33]
Commit https://github.com/faye/websocket-driver-node/commit/85709bb91586d4e313a3abf14f8198375a94ab33#diff-b9cfc7f2cdf78a7f4b91a753d10865a2 introduces a change to package.json that changes a semver range so that it includes a comma character which is not valid according to the [range grammar](https://www.npmjs.com/package/semver#range-grammar) and checking the range is valid using `semver.validRange('>= 0.4.0, < 0.4.11')` returns `null`.

**Comment by jcoglan** [comment]
That's interesting, I didn't know commas were invalid and npm installs the package correctly. Is npm doing something that's a superset of semver?

**Comment by jcoglan** [comment]
For example, npm also appears to allow spaces around operators like `>=` even though that grammar does not allow them.

**Comment by mgroenhoff** [comment]
> That's interesting, I didn't know commas were invalid and npm installs the package correctly. Is npm doing something that's a superset of semver?

I don't know. It might. We have a lot of custom tooling which relies on the semver package which now all fail because it can not resolve the dependency.

> For example, npm also appears to allow spaces around operators like `>=` even though that grammar does not allow them.

I think most whitespace is insignificant and is ignored when parsing. I found this to be the only mention of whitespace in the semver doc: 

> Comparators can be joined by whitespace to form a comparator set, which is satisfied by the intersection of all of the comparators it includes.

**Comment by jcoglan** [comment]
OK, I've yanked 0.7.2 and published 0.7.3 with amended version strings.

**Comment by mgroenhoff** [comment]
> OK, I've yanked 0.7.2 and published 0.7.3 with amended version strings.

Thank you for the quick responses!

I did some more tests and it seems that if you enable [loose mode](https://www.npmjs.com/package/semver#functions) you might not get what you expect. I think this might be what NPM does...

```js
semver.validRange('>= 0.4.0  , < 0.4.11', { loose: true }) // returns '<0.4.11' which is correct but unexpected
semver.validRange('>= 0.4.0    < 0.4.11', { loose: true }) // returns '>=0.4.0 <0.4.11' as expected
```
